### DISCUSSION TITLE
`const` Regex compilation

### DISCUSSION THREAD

**Discussion by sae3023** [rust-lang/regex#1012] [Q&A] [answered]
I think compiling a `&'static str` into a regex is a pretty popular use case.

However, no `const fn make(pattern: &'static str) -> Regex` exists.
Is it impossible to implement? 
Is it possible, but the payout is not high enough, since they are meant to be compiled once anyway?

Sorry if the question is dumb, I'm a noob.

**Comment by BurntSushi** [comment, marked as answer]
It is somewhat addressed here: https://github.com/rust-lang/regex/issues/913

For the specific case of a `const` method, theoretically, one could implement _a_ regex engine using Rust `const`, but certainly not the one in this crate. To a first approximation, in order for this crate to have a `pub const fn new(pattern: &'static str) -> Regex`, you would need to have most of the Rust language usable inside a `const` context. That is *nowhere near* the case today.

The issue I linked addresses the more general case of "compile time regex" and specifically, "a serialization format for a regex that enables cheap deserialization." Rust `const` is just one way of getting a compile time regex. There are other techniques, as that issue covers.
